src/main.ts

```diff
@@ -21,22 +21,28 @@ export async function main(): Promise<Outputs> {
     const deploymentMode = getInput('deploymentMode')
     const deploymentName = getInput('deploymentName')
     const parameters = getInput('parameters')
+    const managementGroupId = getInput('managementGroupId')
+    const validationOnly = getInput('validationOnly') == 'true';
 
     // change the subscription context
-    info("Changing subscription context...")
-    await exec(`"${azPath}" account set --subscription ${subscriptionId}`, [], { silent: true })
+    if (subscriptionId != "") {
+        info("Changing subscription context...")
+        await exec(`"${azPath}" account set --subscription ${subscriptionId}`, [], { silent: true })
+    } else if (scope != "managementgroup") {
+        throw new Error("Subscription Id must be set on Resource Group and Subscription scope.")
```

Comment:
I don't think we should throw here and cause the action to fail.  We can skip setting context if scope is managementGroup. Or just let it proceed after setting context.

Reply:
Shall I print a warning or just do nothing?